src/main/java/qna/service/QnAService.java

```diff
@@ -35,24 +35,50 @@ public Question findQuestionById(Long id) {
     @Transactional
     public void deleteQuestion(User loginUser, long questionId) throws CannotDeleteException {
         Question question = findQuestionById(questionId);
-        if (!question.isOwner(loginUser)) {
-            throw new CannotDeleteException("질문을 삭제할 권한이 없습니다.");
+        List<DeleteHistory> deleteHistories = new ArrayList<>();
+
+        deleteHistories.add(deleteOneQuestion(question, loginUser));
+        deleteHistories.addAll(deleteAnswer(question, loginUser));
+        deleteHistoryService.saveAll(deleteHistories);
+    }
+
+    public void validateIsOwner(User writer, User loginUser) throws CannotDeleteException {
+        if (!writer.matchUserId(loginUser.getUserId())) {
+            throw new CannotDeleteException("질문을 삭제할 수 없습니다.");
         }
+    }
+
+    public DeleteHistory deleteOneQuestion(Question question, User loginUser) throws CannotDeleteException {
+        validateIsOwner(question.getWriter(), loginUser);
+        question.setDeleted(true);
+        return new DeleteHistory(ContentType.QUESTION, question.getId(), question.getWriter(), LocalDateTime.now());
+    }
 
+    public List<DeleteHistory> deleteAnswer(Question question, User loginUser) throws CannotDeleteException {
+        List<DeleteHistory> deleteHistories = new ArrayList<>();
         List<Answer> answers = question.getAnswers();
+
         for (Answer answer : answers) {
-            if (!answer.isOwner(loginUser)) {
-                throw new CannotDeleteException("다른 사람이 쓴 답변이 있어 삭제할 수 없습니다.");
-            }
+            validateIsOwner(answer.getWriter(), loginUser);
         }
 
-        List<DeleteHistory> deleteHistories = new ArrayList<>();
-        question.setDeleted(true);
-        deleteHistories.add(new DeleteHistory(ContentType.QUESTION, questionId, question.getWriter(), LocalDateTime.now()));
+        answers.stream()
+                .forEach(a -> {
```

Comment:
Collection은 기본적으로 forEach를 제공하고 있으니 Stream을 열지 않아도 될것 같아요.
```suggestion
                answers.forEach(a -> {
```